main.tf

```
@@ -1,3 +1,8 @@
+data "aws_subnet" "this" {
+  for_each = length(var.subnet_ids) > 0 ? { for idx, subnet in var.subnet_ids : index => subnet } : {}
```

Simple mis-reference. Commit this fix and I'll re-run tests. 

```suggestion
  for_each = length(var.subnet_ids) > 0 ? { for idx, subnet in var.subnet_ids : idx => subnet } : {}
```